src/aiida/cmdline/commands/cmd_init.py

```diff
@@ -0,0 +1,103 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""``verdi init`` command."""
+import pathlib
+
+import click
+
+from aiida.cmdline.commands.cmd_verdi import verdi
+from aiida.cmdline.utils import echo
+
+
+@verdi.command('init')
+@click.argument(
+    'directory', type=click.Path(exists=False, path_type=pathlib.Path, resolve_path=True), default=pathlib.Path.cwd()
+)
+@click.option(
+    '--from-archive',
+    type=click.Path(dir_okay=False, exists=True, resolve_path=True, path_type=pathlib.Path),
+    help='Mount the archive as a read-only profile instead of creating an empty profile.',
+)
+def verdi_init(directory, from_archive):
+    """Initialize a new AiiDA instance.
+
+    The instance is initialized in DIRECTORY, which defaults to the currrent working directory. A profile is
+    automatically created using the `core.sqlite_dos` storage backend and the localhost is configured as a computer.
+    """
+    import tempfile
+    import warnings
+
+    from aiida.common import exceptions
+    from aiida.manage import get_manager
+    from aiida.manage.configuration import create_profile, get_config, load_profile, reset_config, settings
+    from aiida.orm import Computer
+
+    # Unload the profile that was loaded by default and reset the configuration.
+    get_manager().unload_profile()
+    reset_config()
+
+    dirpath_config = directory / settings.DEFAULT_CONFIG_DIR_NAME
+
+    if dirpath_config.is_dir():
+        echo.echo_critical(f'A configuration directory already exists in `{directory}`.')
+
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore')
+        settings.set_configuration_directory(dirpath_config)
+
+    echo.echo_success(f'Initialized new AiiDA instance in `{dirpath_config}`.')
+
+    # Now load the config causing the configuration file to be created.
+    config = get_config(create=True)
+    dirpath_loaded = pathlib.Path(config.filepath).parent
+    assert dirpath_loaded == dirpath_config, f'Filepath of loaded config `{dirpath_loaded}` != `{dirpath_config}`.'
+
+    if from_archive:
+        storage_backend = 'core.sqlite_zip'
```

Comment:
can a user switch the backend to `sqlite_dos` later easily?

Reply:
No they cannot. But why would they want to do this? If they actually want to use the data in a profile that is mutable, they can simply do the following instead:
```
verdi init
verdi archive import archive.aiida
```
I would say that is simple enough as an alternative.

Reply:
I think a user case: users import an archive using sqlite_zip, then after inspecting the archive, they find it's good to start work directly on top of it, and if there is a simple way to quickly switch the backend, that would be helpful.

Reply:
We cannot really convert the `core.sqlite_zip` backend to an `core.sqlite_dos` one. What we would have to do is just create a new profile from scratch with `core.sqlite_dos` and then import the archive. The command to do this would have to be documented, but in that case I think it is just easier to tell users to do `verdi init && verdi archive import`.